tasks/noauto_create_timer.yml

```diff
@@ -0,0 +1,53 @@
+---
+- name: Create borgbackup timer
+  when:
+    - borgmatic_timer is defined and borgmatic_timer == "systemd"
+  tags:
+    - install_backup
+  block:
+    - name: Copy systemd files
+      ansible.builtin.template:
+        src: "{{ item.src }}"
+        dest: "{{ item.dest }}"
+        owner: root
+        group: root
+        backup: true
+        mode: "{{ item.mode }}"
+      with_items:
+        - { src: "backup.timer.j2", dest: "/usr/lib/systemd/system/backup.timer", mode: "0644" }
+        - { src: "backup.service.j2", dest: "/usr/lib/systemd/system/backup.service", mode: "0644" }
+
+    - name: Populate service facts
+      ansible.builtin.service_facts:
+
+      # If the role is running and the repo is not yet initialized, an error will occur.
+      # Therefore the service is stopped by default and must be started manually.
+    - name: Stop fresh installed backup.timer and backup.service
+      when: ('backup.service' not in services)
+      block: 
+        - name: Stop borgmatic services
+          ansible.builtin.systemd:
+            name: "{{ item }}"
+            state: stopped
+            enabled: false
+            masked: false
+            daemon_reload: true
+          when: "item in services"
```

Comment:
This fails tests currently. Maybe you need `ansible_facts.services`?

```
  TASK [m3nu.ansible_role_borgbackup : Stop borgmatic services] ******************
  fatal: [almalinux-8]: FAILED! => {"msg": "The conditional check '('backup.service' not in services)' failed. The error was: error while evaluating conditional (('backup.service' not in services)): 'services' is undefined. 'services' is undefined\n\nThe error appears to be in '/home/runner/work/ansible-role-borgbackup/ansible-role-borgbackup/tasks/noauto_create_timer.yml': line 28, column 11, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n      block:\n        - name: Stop borgmatic services\n          ^ here\n"}
```

Reply:
I have a lot of trouble this dist on my tests, too.
Sometimes it works an sometimes not.

I installed today an amarlinux8 and everything works fine.

If you can reproduce this issue, please add under

    - name: Populate service facts
      ansible.builtin.service_facts:

this lines:

    - name: Print service facts
      ansible.builtin.debug:
        var: ansible_facts.services

Reply:
This needs to work on all platforms we test. I'll try a few things.

Reply:
Tests fail here because we test inside Docker and not a full VM.

Reply:
Can i close that conversation-part? Or have we a doing inside the restructure?

Reply:
Yes, this is a limitation of Molecule. So we can only use Cron during tests. I'll test it in a VM later.